packages/form-builder/src/blocks/fields/amount/Edit.tsx

```diff
@@ -1,51 +1,201 @@
+    const isRecurringAdmin = recurringEnabled && 'admin' === recurringDonationChoice;
+    const isRecurringDonorPreset =
+        recurringEnabled && 'donor' === recurringDonationChoice && 'preset' === recurringPeriod;
+    const isRecurringDonorChoice =
+        recurringEnabled && 'donor' === recurringDonationChoice && 'donor' === recurringPeriod;
+    const amountFormatted = formatCurrencyAmount(setPrice);
+
+    const DonationLevels = () => (
+        <LevelGrid>
+            {levels.map((level, index) => {
+                const levelAmount = formatCurrencyAmount(level);
+
+                return <LevelButton key={index}>{levelAmount}</LevelButton>;
+            })}
+        </LevelGrid>
+    );
+
+    const CustomAmount = () => (
+        <CurrencyControl value={setPrice} label={__('Custom amount', 'give')} hideLabelFromVision />
+    );
+
+    const FixedPriceMessage = () => (
+        <Notice>
+            {createInterpolateElement(__('This donation is set to <amount/> for this form.', 'give'), {
+                amount: <strong>{amountFormatted}</strong>,
+            })}
+        </Notice>
+    );
+
+    const RecurringPeriod = ({count, label}) => {
+        const interval = count ?? recurringBillingInterval;
+
+        const singular = !isRecurringDonorChoice
+            ? periodLookup[recurringBillingPeriod].singular
+            : periodLookup[recurringBillingPeriodOptions[0]].singular;
+
+        const plural = !isRecurringDonorChoice
+            ? periodLookup[recurringBillingPeriod].plural
+            : periodLookup[recurringBillingPeriodOptions[0]].plural;
+
+        return (
+            <strong>
+                {1 === interval && <>{label ?? singular}</>}
+                {1 !== interval && (
+                    <>
+                        {interval} {label ?? plural}
+                    </>
+                )}
+            </strong>
+        );
+    };
 
-    const FixedPriceMessage = () => {
-        const amount = formatCurrencyAmount(setPrice);
+    const RecurringMessage = () => {
+        const interval = parseInt(recurringBillingInterval);
 
-        return createInterpolateElement(__('This donation is set to <amount/> for this form.', 'give'), {
-            amount: <strong>{amount}</strong>,
-        });
+        const periodPlaceholder = <strong>{'[' + _n('period', 'periods', interval, 'give') + ']'}</strong>;
+
+        const periodElement = !isRecurringDonorChoice ? (
+            <RecurringPeriod count={interval} label={periodPlaceholder} />
+        ) : (
+            <RecurringPeriod count={interval} label={periodPlaceholder} />
+        );
+
+        const count = parseInt(recurringLengthOfTime);
+        const countElement = <strong>{count}</strong>;
+
+        const translatableString = !count
+            ? __('This donation occurs every <period />.', 'give')
+            : __('This donation occurs every <period /> for <count /> <strong>payments</strong>.', 'give');
+
```

Comment:
This was a simplification, showing a placeholder. I think this comes from the design. I had looked into making this dynamic, but that complicated the implementation (seemingly unnecessarily).

Reply:
The design _does_ show the period value when the donation choice is admin though.  We can keep the redacted period for when it's donors choice because we are not interacting with the periods in the preview.

<img width="801" alt="Screenshot 2023-03-02 at 12 06 24 PM" src="https://user-images.githubusercontent.com/10138447/222500717-8410e049-bb76-4100-a272-8ee794424340.png">

Reply:
Resolved, see https://github.com/impress-org/givewp-next-gen/pull/128/commits/31d906b6edd732b3fcfaeafd0877cd7857927ce4.